game_board.py

```diff
@@ -148,60 +148,120 @@ def which_color_pair(stone_marker):
         color = 3
     return color
 
-def move_down(plus_list,current_row):
-    """ Finds the "+" char that is below the current_row. If the current "+" char is at the bottom, 
-        it finds the "+" char at the top with the same x position
-    Returns the new y position of the "+" 
+def move_down(plus_list,loc):
+    """ Finds the "+" char that is above loc. If the current "+" char is at a coordinate that
+        that violates the rules for movement, the positiono wont change.
+    Returns the new x position of the "+"  
 
-    Keyword arguments:
-    current_row -- the currently selected row in the plus_list
+
+    loc -- the currently selected position in the plus_list
     plus_list -- the lists of all the "+" and their positions
     """
-    
-    new_row = current_row
+    new_loc = loc
     
     while True:
-        new_row += 1
-        if new_row == len(plus_list):
-            new_row = 0
+        new_loc += 1
+        if new_loc == len(plus_list):
+            new_loc = 0
         
-        current_x = plus_list[current_row][1]
-        new_x = plus_list[new_row][1]
+        current_x = plus_list[loc][1]
+        new_x = plus_list[new_loc][1]
 
-        current_y = plus_list[current_row][2]
-        new_y = plus_list[new_row][2]
+        current_y = plus_list[loc][2]
+        new_y = plus_list[new_loc][2]
         if current_x == new_x and current_y != new_y:
-          return new_row 
+            #Check if allowed to move down
+            if (int(current_x) != 16 and int(current_y) <= 6) or (int(current_x) == 16 and int(current_y) != 4 and int(current_y) != 12):
```

Comment:
What is so special about the numbers 16, 4 and 12? Why do we need to check against these specific values. I would suggest putting them into variables that explain what they are or perhaps they can be calculated?

Reply:
I made two new variables 'middle' and 'step' that represents the middle of the map and the length of a "step" along the x- and y-axis.